### ISSUE TITLE
greenlet.error: cannot switch to a different thread

### ISSUE THREAD

**Issue by phmarek** [neovim/pynvim#254]
The python child processes crash (more often than not) with this error:

```
12852 08:57:04.770080 write(1, "\224\1\0\332\2&Traceback (most recent call last):\n
  File \"/usr/local/lib/python3.5/dist-packages/neovim/msgpack_rpc/async_session.py\", line 75, in _on_message\n
    self._handlers.get(msg[0], self._on_invalid_message)(msg)\n
  File \"/usr/local/lib/python3.5/dist-packages/neovim/msgpack_rpc/async_session.py\", line 96, in _on_response\n
    self._pending_requests.pop(msg[1])(msg[2], msg[3])\n
  File \"/usr/local/lib/python3.5/dist-packages/neovim/msgpack_rpc/session.py\", line 150, in response_cb\n
    gr.switch(err, rv)\n
greenlet.error: cannot switch to a different thread\n
\300", 557 <unfinished ...>
```

This results in the famous  `Invalid channel "1"` error message in neovim, which is **very** annoying.

See also https://github.com/neovim/neovim/issues/6249; but it seems to be related with the python client, so I'm copying the error here.

**Comment by bfredl** [comment]
Please try #255. There would still be an error, but hopefully a more helpful traceback and not crashing the client.

**Comment by mcmire** [comment]
Just ran into this issue. Would love there to be a solution to this as Python support in Neovim seems broken otherwise.

**Comment by justinmk** [comment]
Addressed by #255
